spec/lib/cap/authors_poller_spec.rb

```diff
@@ -223,4 +223,25 @@
       expect { subject.process_next_batch_of_authorship_data(bogus: 'data') }.to raise_error(Net::HTTPBadResponse)
     end
   end
+
+  describe '.do_science_wire_harvest' do
+    it 'does nothing if sw client is disabled' do
+      current_sw_setting = Settings.SCIENCEWIRE.enabled
+      Settings.SCIENCEWIRE.enabled = false
+      subject.do_science_wire_harvest
+      expect(ScienceWireHarvester).not_to receive(:harvest_pubs_for_author_ids)
+      Settings.SCIENCEWIRE.enabled = current_sw_setting
+    end
+  end
+
+  describe '.do_wos_harvest' do
+    it 'does nothing if WoS client is disabled' do
+      current_wos_setting = Settings.WOS.enabled
+      Settings.WOS.enabled = false
+      subject.do_wos_harvest
+      expect(Author).not_to receive(:where)
+      expect(WebOfScience.harvester).not_to receive(:harvest)
+      Settings.WOS.enabled = current_wos_setting
+    end
+  end
```

Comment:
:+1: to adding specs for this :balloon: 

Consider using `allow(Settings.WOS).to receive(:enabled).and_return(true)` (or false) to mock setting these values.

Reply:
It might simplify to `expect(WebOfScience).not_to receive(:harvester)`

Reply:
I usually put an `expect` before calling something to ensure it's mocked, e.g.
```
allow(Settings.WOS).to receive(:enabled).and_return(false)
expect(WebOfScience).not_to receive(:harvester)
subject.do_wos_harvest
```

Reply:
thanks for the suggestions, the mocking of settings is a much better way of doing that... all made, rebased and pushed